spec/models/validations/shared_validations_spec.rb

```diff
@@ -71,10 +71,11 @@
 
     context "when validating percent" do
       it "validates that % suffix is added in the error message" do
-        sales_record.stairbought = "random"
+        question_under_test = Form::Sales::Questions::StaircaseBought.new(nil, nil, self)
+        sales_record.stairbought = question_under_test.max + 1
         shared_validator.validate_numeric_min_max(sales_record)
         expect(sales_record.errors["stairbought"])
-          .to include(match I18n.t("validations.numeric.valid", field: "Percentage bought in this staircasing transaction", min: "0 percent", max: "100 percent"))
+          .to include(match I18n.t("validations.numeric.valid", field: question_under_test.check_answer_label, min: "#{question_under_test.min}#{question_under_test.suffix}", max: "#{question_under_test.max}#{question_under_test.suffix}"))
```

Comment:
@kosiakkatrina let me know what you think of this test.
I think it also has the advantage of having fewer magic numbers and strings floating around in it that could theoretically need updating in future when the structure of the test could still be valid.
If we like this form generally that I will investigate converting 75 to a public_send to pull the "stairbought" magic string out even a touch more.
I can then amend some of the other test in this file to follow this pattern.

Reply:
I think I'd keep it the way it was because then we'd be testing that `question_under_test.check_answer_label` and min/max methods get called correctly and return what we expect as well as testing that the error got added

Reply:
I think this specifically doesn't care what the max is, it just cares that we set the field to be one greater than whatever the max is.
The way I see it either formulation makes the assumption that the question for the the field has a max set, and writing the test with a lot of hard-coded numbers or strings makes more assumptions that those numbers in the question are what we think they are.

Reply:
How do you know that `question_under_test.max`, `question_under_test.suffix` and `question_under_test.check_answer_label` aren't nil?